data/datasource.go

```diff
@@ -426,6 +431,36 @@ func readConsul(source *Source, args ...string) ([]byte, error) {
 	return data, nil
 }
 
+func readK8S(source *Source, args ...string) ([]byte, error) {
+	if len(args) < 1 {
+		return nil, fmt.Errorf("a key is needed")
+	}
+	if source.K8S == nil {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+		// creates the clientset
+		source.K8S, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, err
+		}
+	}
+	namespace := source.URL.Host
+	configmapName := strings.Trim(source.URL.Path, "/")
+	configmap, err := source.K8S.CoreV1().ConfigMaps(namespace).Get(configmapName, meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if data, ok := configmap.Data[args[0]]; ok {
+		return []byte(data), nil
+	}
+	if len(args) > 1 {
+		return []byte(args[1]), nil
```

Comment:
The `datasource` function doesn't support default values, and I'd rather not add that support to this specific datasource - so please remove this block.

There was some discussion of a `default` function in #213 that would solve the general case.